### PR TITLE
Rebrand on upgrade

### LINKED ISSUE #928: GRUB menu rebranding doesn't work after upgrading
**cos-toolkit version:**
utils/installer-0.21.1

**CPU architecture, OS, and Version:**
Linux node1 5.3.18-59.34-default #1 SMP Thu Nov 11 12:18:45 UTC 2021 (a2a53aa) x86_64 x86_64 x86_64 GNU/Linux

**Describe the bug**
In Harvester we add `GRUB_ENTRY_NAME="Harvester {VERSION}` to `/etc/os-release` to display the name in GRUB menu entries. e.g.,
```
NAME="Harvester"
ID="harvester"
ID_LIKE="suse opensuse"
VERSION_ID="20211130"
PRETTY_NAME="Harvester 1.0.0-rc1"
ANSI_COLOR="0;32"
BUG_REPORT_URL="https://github.com/harvester/os2/issues"
HOME_URL="https://github.com/harvester/os2"
GRUB_ENTRY_NAME="Harvester 1.0.0-rc1"
```

After upgrading the OS with a new image by calling `cos upgrade`, the GRUB menu entry still use the old one:

```
node3:~ # cat /etc/os-release
NAME="Harvester"
ID="harvester"
ID_LIKE="suse opensuse"
VERSION_ID="20211130"
PRETTY_NAME="Harvester b62f9f8-dirty"
ANSI_COLOR="0;32"
BUG_REPORT_URL="https://github.com/harvester/os2/issues"
HOME_URL="https://github.com/harvester/os2"
GRUB_ENTRY_NAME="Harvester b62f9f8-dirty"

node3:~ # cat /run/initramfs/cos-state/grub_oem_env
# GRUB Environment Block
# WARNING: Do not edit this file other than by grub2-editenv
default_menu_entry=Harvester v1.0.0-rc1
##################################################################################################################################################################################################################################################################################################################################################################################################################################################################################################################################################################################################################################################################################################################################################################################################################################################################################################################################
```
**To Reproduce**

Upgrade a cOS derivative with a new image which has a different `GRUB_ENTRY_NAME` option in `/etc/os-release`.
**Expected behavior**

The GRUB menu should be rebranded.

**Logs**

**Additional context**

- Looks like `rebrand` is not performed in `upgrade` path.
- I can workaround by remounting `/run/initramfs/cos-state/`  in rw mode and run the `grub2-editenv /run/boot/grub_oem_env set default_menu_entry="$grub_entry"` command.

### PR DESCRIPTION
**Pull request by mudler**
This PR calls `cos-rebrand` among the chroot hooks in order to apply branding from the system which is being upgraded.
The chroot hooks are called after the target is set up so this ensures we have the target ready, while probably this isn't the best place for it, we can keep this in mind while working on the new cli. Aside, as we have to backport this, I wanted to keep the changes minimal.

It adds a simple test inside the upgrade one instead of a separate one in order to not overload the CI

Fixes #928

### CODE REVIEW

**Review by Itxaka — APPROVED** [review]

**Review by dragonchaser — APPROVED** [review]
lgtm

**Comment by mudler** [comment]
Seems there is still something not right with it, having a closer :eyes:

**Comment by mudler** [comment]
I've just tested this manually and it is successful. I'm thinking that it fails here just because it upgrades to an image that is missing https://github.com/rancher-sandbox/cOS-toolkit/pull/959/files#diff-51293a201a1e5be4d965ec46c9d5d9e012b94efd711b35a478e09ad03818a506R1212 , which actually is still the case on the older installer

**Comment by mudler** [comment]
As a side note, the backported installer already did call the configuration loading here: https://github.com/rancher-sandbox/cOS-toolkit/blob/master/packages/backports/installer/cos.sh#L1040 so this looks like a regression in the version we currently have in master. 

I'll try to set up the test separately and make it run against upgrading a new image which contains the above change. I will mark it at pending, and later enable it after we have merged this in master

**Comment by mudler** [comment]
So, I'm getting space left on device, but the funny thing is that the runner then starts to fail badly, and as we cleanup right after the error seem to  be lost somehow. Hard to tell from the logs, because seems GH runner just goes desync (maybe for the I/O space?)

```
2021-12-14T12:51:18.6408528Z        #12 exporting to image
2021-12-14T12:51:18.6409667Z        #12 sha256:e8c613e07b0b7ff33893b694f7759a10d42e180f2b4dc349fb57dc6b71dcab00
2021-12-14T12:51:18.6410610Z        #12 exporting layers
2021-12-14T12:51:25.9154195Z  INFO   #12 exporting layers 7.4s done
2021-12-14T12:51:25.9158521Z  INFO   #12 writing image sha256:2dfbee62e99e53bfdc96ca648523eae46cbae1d4a4dd48a59e637e913978a389 done
2021-12-14T12:51:25.9166429Z  INFO   #12 naming to quay.io/costoolkit/build-green-cache:6e6f600449ce61511d9f306679d1fb87de41c6fc7a281d5ce003f3a20bc7e69b done
2021-12-14T12:51:25.9171371Z  INFO   #12 DONE 7.4s
2021-12-14T12:51:25.9208870Z  SUCCESS    Building image quay.io/costoolkit/build-green-cache:6e6f600449ce61511d9f306679d1fb87de41c6fc7a281d5ce003f3a20bc7e69b done
2021-12-14T12:51:41.1285283Z   ERROR    Error: building direct image: Error met while extracting image: Failed saving image: Error response from daemon: write /var/lib/docker/tmp/docker-export-925525055/35c40d8e3ad8fd455f4cbcc8b9b99b3da547ef9ce1883df1b42d8973f0a900eb/layer.tar: no space left on device
2021-12-14T12:51:41.1288008Z           : exit status 1
2021-12-14T12:51:41.1288704Z   ERROR    Bailing out
2021-12-14T12:51:41.1425653Z make: *** [make/Makefile.build:86: build] Error 1
2021-12-14T12:51:41.3390171Z Untagged: quay.io/costoolkit/build-green-cache:6e6f600449ce6
```

So the build ended up without a new recovery img, hence why tests for squashfs fail. Going to retry again, but we might have to look at #939 sooner than expected

**Comment by mudler** [comment]
Anyhow, the https://github.com/rancher-sandbox/cOS-toolkit/runs/4520831105?check_suite_focus=true unsigned passed on non-squashfs, and should make no difference in that case as it does not involve recovery (which tests are passing too. https://github.com/rancher-sandbox/cOS-toolkit/runs/4520830924?check_suite_focus=true ). So this should be good to go

**Comment by bk201** [comment]
@mudler Thanks for the fix and backport.  I can't install `utils/installer-0.21.3`, did I miss anything? 
```
Step 24/29 : RUN luet install -y     toolchain/yip     toolchain/luet     utils/installer@0.21.3     system/cos-setup     system/immutable-rootfs@0.2.0-12     system/grub2-config     selinux/k3s     selinux/rancher     utils/k9s     utils/nerdctl     toolchain/yq
 ---> Running in f3dd5b172ad6
|  Downloading quay.io/costoolkit/releases-green:repository.yaml
-  Pulled: sha256:7bfa4bc53c4c7060f6d138e3e3957fda231f96eb29bb45bc48c82234575b458a
 Size: 628B
 Downloading quay.io/costoolkit/releases-green:tree.tar.zst
\  Pulled: sha256:9fd90961be03cbce1289b01fe478ee008de83c46b64e8c58e2086b049e84c003
 Size: 12.48KiB
 Downloading quay.io/costoolkit/releases-green:repository.meta.yaml.tar.zst
|  Pulled: sha256:5cbf0e3e777e8913f46c5784fe46619046dfadcdd8f4aa9c2f6cb86521a5d22d
 Size: 819.3KiB
-   Repository cOS revision: 188 - 2021-12-14 0903 +0000 UTC
\  Repository: cos Priority: 1 Type: docker
 Error: Failed matching solutions against repository for utils/installer-0.21.3 where are definitions coming from?!
The command '/bin/sh -c luet install -y     toolchain/yip     toolchain/luet     utils/installer@0.21.3     system/cos-setup     system/immutable-rootfs@0.2.0-12     system/grub2-config     selinux/k3s     selinux/rancher     utils/k9s     utils/nerdctl     toolchain/yq' returned a non-zero code: 1
```

**Comment by mudler** [comment]
> @mudler Thanks for the fix and backport. I can't install `utils/installer-0.21.3`, did I miss anything?
> 
> ```
> Step 24/29 : RUN luet install -y     toolchain/yip     toolchain/luet     utils/installer@0.21.3     system/cos-setup     system/immutable-rootfs@0.2.0-12     system/grub2-config     selinux/k3s     selinux/rancher     utils/k9s     utils/nerdctl     toolchain/yq
>  ---> Running in f3dd5b172ad6
> |  Downloading quay.io/costoolkit/releases-green:repository.yaml
> -  Pulled: sha256:7bfa4bc53c4c7060f6d138e3e3957fda231f96eb29bb45bc48c82234575b458a
>  Size: 628B
>  Downloading quay.io/costoolkit/releases-green:tree.tar.zst
> \  Pulled: sha256:9fd90961be03cbce1289b01fe478ee008de83c46b64e8c58e2086b049e84c003
>  Size: 12.48KiB
>  Downloading quay.io/costoolkit/releases-green:repository.meta.yaml.tar.zst
> |  Pulled: sha256:5cbf0e3e777e8913f46c5784fe46619046dfadcdd8f4aa9c2f6cb86521a5d22d
>  Size: 819.3KiB
> -   Repository cOS revision: 188 - 2021-12-14 0903 +0000 UTC
> \  Repository: cos Priority: 1 Type: docker
>  Error: Failed matching solutions against repository for utils/installer-0.21.3 where are definitions coming from?!
> The command '/bin/sh -c luet install -y     toolchain/yip     toolchain/luet     utils/installer@0.21.3     system/cos-setup     system/immutable-rootfs@0.2.0-12     system/grub2-config     selinux/k3s     selinux/rancher     utils/k9s     utils/nerdctl     toolchain/yq' returned a non-zero code: 1
> ```

Nothing on your side, just waiting for CI on master to be green. Currently failing on pushing cosign signatures https://github.com/rancher-sandbox/cOS-toolkit/runs/4534197096?check_suite_focus=true

**Pull request merged**
